Build instruction

### Describe the problem to be solved

we are trying to testing something for performance issue and want to add more logs for monitor status. Could you please kindly share the instruciton how to build spegel project, so we can rebuild code changing and testing it before check in to upstream. 

### Proposed solution to the problem

_No response_

run make docker-build, got the error below:
```
=> ERROR [builder 6/9] RUN go mod download                                                                                                         30.4s
------
 > [builder 6/9] RUN go mod download:
30.38 go: github.com/AdaLogics/go-fuzz-headers@v0.0.0-20230811130428-ced1acdcaa24: Get "https://proxy.golang.org/github.com/%21ada%21logics/go-fuzz-headers/@v/v0.0.0-20230811130428-ced1acdcaa24.mod": dial tcp: lookup proxy.golang.org: i/o timeout
------
Dockerfile:6
--------------------
   4 |     COPY go.mod go.mod
   5 |     COPY go.sum go.sum
   6 | >>> RUN go mod download
   7 |     COPY main.go main.go
   8 |     COPY pkg/ pkg/
--------------------
ERROR: failed to solve: process "/bin/sh -c go mod download" did not complete successfully: exit code: 1
make: *** [Makefile:13: docker-build] Error 1
```

OS Indformation:
```
» lsb_release -a                                                                           
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.3 LTS
Release:	22.04
Codename:	jammy
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [#339](https://github.com/XenitAB/spegel/pull/339) Extend OCI client tests.
 - [#339](https://github.com/XenitAB/spegel/pull/339) Extend OCI client tests.
 - [#365](https://github.com/XenitAB/spegel/pull/365) Add support for throttling blob write speed.
+- [#386](https://github.com/XenitAB/spegel/pull/386) Add contributing guide.
 
 ### Changed
 

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -1,0 +1,62 @@
+# Contributing
+
+Thank you for considering contributing to Spegel, hopefully this document will make this process easier.
+
+## Running tests
+
+The following tools are required to run the tests properly.
+
+* go
+* golangci-lint
+* kind
+
+Run the linter and the unit tests to quickly validate changes.
+
+```shell
+make lint test
+```
+
+Run the e2e tests which take a bit more time.
+
+```shell
+make e2e
+```
+
+There are e2e tests for the different CNIs iptables, iptables-v6, and ipvs.
+
+```shell
+make e2e CNI=ipvs
+```
+
+## Building
+
+Build the Docker image locally.
+
+```shell
+make docker-build
+```
+
+It is possible to specify a different image name and tag.
+
+```shell
+make docker-build IMG=exmaple.com/spegel TAG=feature
+```
+
+## Generating documentation
+
+Changes to the Helm chart values will require the documentation to be regenerated.
+
+```shell
+make helm-docs
+```
+
+## Acceptance policy
+
+Pull requests need to fulfill the following requirements to be accepted.
+
+* New code has tests where applicable.
+* The change has been added to the [changelog](./CHANGELOG.md).
+* Documentation has been generated if applicable.
+* The unit tests pass.
+* Linter does not report any errors.
+* All end to end tests pass.

